clients/client/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java

```diff
@@ -0,0 +1,202 @@
+  }
+
+  @ParameterizedTest
+  @MethodSource("negativeCases")
+  void testNegativeSerDeCases(Case invalidCase) {
+    assertThrows(
+        JsonProcessingException.class,
+        () -> MAPPER.readValue(invalidCase.deserializedJson, invalidCase.deserializeAs));
+  }
+
+  static List<Case> goodCases() {
+    final Instant now = Instant.now();
+    final String branchName = "testBranch";
+
+    return Arrays.asList(
+        new Case(
+            ImmutableBranch.builder().hash(HASH).name(branchName).build(),
+            Branch.class,
+            "{\"type\":\"BRANCH\",\"name\":\"testBranch\"," + "\"hash\":\"" + HASH + "\"}"),
+        new Case(
+            ImmutableBranch.builder().name(branchName).build(),
+            Branch.class,
+            "{\"type\":\"BRANCH\",\"name\":\"testBranch\",\"hash\":null}"),
+        new Case(
+            ImmutableTag.builder().hash(HASH).name("tagname").build(),
+            Tag.class,
+            "{\"type\":\"TAG\",\"name\":\"tagname\",\"hash\":\"" + HASH + "\"}"),
+        new Case(
+            ImmutableHash.builder().name(HASH).build(),
+            Hash.class,
+            "{\"type\":\"HASH\",\"name\":\"" + HASH + "\",\"hash\":\"" + HASH + "\"}"),
+        new Case(
+            ImmutableTransplant.builder()
+                .addHashesToTransplant(HASH)
+                .fromRefName(branchName)
+                .build(),
+            Transplant.class,
+            "{\"hashesToTransplant\":" + "[\"" + HASH + "\"]," + "\"fromRefName\":\"testBranch\"}"),
+        new Case(
+            ImmutableEntriesResponse.builder()
+                .addEntries(
+                    ImmutableEntry.builder()
+                        .type(Type.ICEBERG_TABLE)
+                        .name(ContentsKey.fromPathString("/tmp/testpath"))
+                        .build())
+                .token(HASH)
+                .hasMore(true)
+                .build(),
+            EntriesResponse.class,
+            "{\"hasMore\":true,"
+                + "\"token\":\""
+                + HASH
+                + "\","
+                + "\"entries\":[{\"type\":\"ICEBERG_TABLE\","
+                + "\"name\":{\"elements\":[\"/tmp/testpath\"]}}]}"),
+        new Case(
+            ImmutableLogResponse.builder()
+                .addOperations()
+                .token(HASH)
+                .addOperations(
+                    ImmutableCommitMeta.builder()
+                        .commitTime(now)
+                        .author("author@testnessie.com")
+                        .committer("committer@testnessie.com")
+                        .authorTime(now)
+                        .hash(HASH)
+                        .message("test commit")
+                        .putProperties("prop1", "val1")
+                        .signedOffBy("signer@testnessie.com")
+                        .build())
+                .hasMore(true)
+                .build(),
+            LogResponse.class,
+            "{\"hasMore\":true,"
```

Comment:
I think, this can be more written more compact, especially when using `String.format()` for `HASH` and `now`.